extmod/network_cyw43.c

```diff
@@ -280,13 +292,19 @@ STATIC mp_obj_t network_cyw43_connect(size_t n_args, const mp_obj_t *pos_args, m
     if (ret != 0) {
         mp_raise_OSError(-ret);
     }
+    #ifdef hal_set_network_active
+    hal_set_network_active(MP_QSTR_WLAN, true);
+    #endif
```

Comment:
As above, I don't think this is needed.

Reply:
I'm going to disagree. We need some sort of check in here for network.STA_IF. All active(True) does for STA is load the driver. And active(False) does nothing for STA. It's connect and disconnect that matters.

Reply:
Yes you're right.  There really needs to be a check in connect/disconnect that it should fail if the interface is not activated.

But that's a bug to fix at a later stage.  For now I think what you've done is correct.